### ISSUE TITLE
hui-element: style is unstable

### ISSUE THREAD

**Issue by ildar170975** [thomasloven/lovelace-card-mod#276]
My Home Assistant version: 2023.4.0

My lovelace configuration method (GUI or yaml): yaml/storage

What I am doing:
I am trying to apply a style for `hui-element` card.
Since this card does not have `ha-card`, it is placed into `mod-card`:
```
type: entities
entities:
  - type: custom:mod-card
    card:
      type: custom:hui-element
      row_type: section
    card_mod:
      style:
        hui-element:
          $: |
            .divider {
              background-color: red !important;
            }
```
A correct output is:
![image](https://user-images.githubusercontent.com/71872483/231635010-69d9e33e-46fd-4843-9990-a26505239fe7.png)

Recently (probably after 2023.4) the style became **unstable** - in most cases it is not applied.

Why doing this test? I am using that code inside a decluttering-card to add a divider line inside stacks (like `stack-in-card`).

What I expected to happen:
Style is always applied.

What happened instead:
Style usually is not applied, but sometimes applied.

**Minimal** steps to reproduce:
as above

Error messages from the browser console:
none

---

**By putting an X in the boxes ([]) below, I indicate that I:**

- [x] Understand that this is a channel for reporting bugs, not a support forum (https://community.home-assistant.io/).

- [x] Have made sure I am using the latest version of the plugin.

- [x] Have followed the troubleshooting steps of the "Common Problems" section of https://github.com/thomasloven/hass-config/wiki/Lovelace-Plugins.

- [x] Understand that leaving one or more boxes unticked or failure to follow the template above may increase the time required to handle my bug-report, or cause it to be closed without further action.

**Comment by ildar170975** [comment]
Same behaviour in Chrome+Win, iOS companion app:
1. In storage mode - style may BE, may BE NOT applied after F5 or refreshing a page.
2. In yaml mode - style IS applied after calling a "Refresh" menu (3dots menu on the top-right corner) and is NEVER applied after F5 or refreshing a page.

**Comment by boardlord1** [comment]
I think my issue is connected to this.

I've used the Mini Graph Card custom card to displace temps in my rooms. I've also used card-mod to make the borders of these cards red if the heating is on in them. Before 2023.4, these worked beautifully.

Now with 2023.4 if the heating comes on, the red border often doesn't show up. Multiple page refreshes, cache clearings later I could usually make it appear, but not always.

Example:
![image](https://user-images.githubusercontent.com/8179205/232452250-70d9ca16-53f7-4d73-adbf-738d15709ee5.png)

My card-mod css code works for sure, as it used to work, and if I go into the edit the card dialogue (while on the main dashboard the red border isn't showing up), the preview shows the red border
Example:
![image](https://user-images.githubusercontent.com/8179205/232452333-1c97e902-2b25-4783-ab6f-d3264c033d0e.png)

YAML code of the card:
```
square: false
columns: 2
type: grid
cards:
  - type: custom:mini-graph-card
    name: Living Room
    entities:
      - entity: sensor.average_mi_li_room_temperature
    line_color: blue
    card_mod:
      style: |
        ha-card {
          border:
            {%- set sensor = states('input_boolean.living_room_temp_vs_heating_set_point') %}
            {%- set sensor2 = states('binary_sensor.living_room_door_sensor_contact') %}
            {%- if sensor == off and sensor2 == 'off' %}
            {%- elif sensor == 'on' and sensor2 == 'off' %}solid 3px red
            {%- elif sensor == 'off' and sensor2 == 'on' %}solid 3px white
            {%- else %}
            {%- endif %} 
  - type: custom:mini-graph-card
    name: Bedroom
    entities:
      - entity: sensor.average_mi_bedroom_temperature
    line_color: dodgerblue
    card_mod:
      style: |
        ha-card {
          border:
            {%- set sensor = states('input_boolean.bedroom_temp_vs_heating_set_point') %}
            {%- set sensor2 = states('binary_sensor.bedroom_window_sensor_contact') %}
            {%- if sensor == off and sensor2 == 'off' %}
            {%- elif sensor == 'on' and sensor2 == 'off' %}solid 3px red
            {%- elif sensor == 'off' and sensor2 == 'on' %}solid 3px white
            {%- else %}
            {%- endif %}
```
But if I close the preview, nothing changes. If I change the code, save, change it back to how it was and save again, the red border shows up on the dashboard

The situation is the same with the Android companion, only after multiple force closes can I make the borders appear. After this it works for the next few changes, then I've reload/force close again to make it update the dashboard.

I've created an issue for the HA Frontend, but seems this is because of card-mod and Home Assistant 2023.4.x?

**Comment by ildar170975** [comment]
Your card has no `hui-element`.
Please create a separate issue for this.

**Comment by ildar170975** [comment]
> card-mod styling is not showing up on the dashboard,

Please create a separate issue.

**Comment by boardlord1** [comment]
Feels like duplicating, but ok...

As for border, I'll add "solid 1px var(--ha-card-border-color,var(--divider-color,#e0e0e0))"

**Comment by ildar170975** [comment]
Could be duplicating, could be not.
Now I see that MY code reproducing YOUR card works FLAWLESS.
That is why I asked you several posts ago - "create separate issue".
Now this thread is filled with irrelevant posts.

**Comment by DATAstrm** [comment]
Just want to chime in that I'm also having issues where my layout will change slightly.  The correct layout seems to always result when I update the Yaml of a view and click the refresh button that pops up on the bottom of the page.  When I do a full page refresh, the layout is slightly different.  It sounds similar to the issue here.

**Comment by ildar170975** [comment]
@DATAstrm 
Is it with `hui-element`?
If no - please create a separate issue.

**Comment by DATAstrm** [comment]
It's not.  It happens with the mini-media player mod so I'm unable to
determine whether it's this addon or that one.

And I have no real way of isolating the error further because it only seems
to happen on my tablet/panel monitor.  Just chiming in, hoping that the
issues are related.  Thanks.

**Comment by ildar170975** [comment]
> And I have no real way of isolating the error further because it only seems to happen on my tablet/panel monitor.

When you create a separate issue - I will come and see if I can help.

**Comment by propi62** [comment]
i have a similar problem. hoped a recent update would fix this:
![image](https://github.com/thomasloven/lovelace-card-mod/assets/39047802/5ddc5835-a09e-44f3-b80d-f9539fe64ae1)

**Comment by ildar170975** [comment]
@propi62 
Is it with hui-element?
If no - please create a separate issue.

**Comment by ildar170975** [comment]
Starting from 3.4.0 it became worse.
Both styles do not work in general:
```
type: entities
entities:
  - type: custom:mod-card
    card:
      type: custom:hui-element
      row_type: section
    card_mod:
      style:
        hui-element:
          $: |
            .divider {
              background-color: red !important;
            }
  - type: custom:mod-card
    card:
      type: custom:hui-element
      row_type: section
    card_mod:
      style:
        hui-element $: |
          .divider {
            background-color: red !important;
          }
```
The 2nd style works sometimes after F5. The 1st style - never worsks.

**Comment by ildar170975** [comment]
In fact this issue is about the same as closed https://github.com/thomasloven/lovelace-card-mod/issues/177

**Comment by ildar170975** [comment]
2024.10
Testing the [card](https://github.com/thomasloven/lovelace-card-mod/issues/276#issuecomment-1886170260):
1. Storage mode:
- the 1st style does not work in Editor, sometimes works after F5 (usually not)
- the 2nd style works stably
2. Yaml mode:
- the 1st style sometimes works after F5 (usually not)
- the 2nd style works stably

**Comment by ildar170975** [comment]
Up until being resolved in a release version.
Still see it with 3.4.5 in 2025.10.1

**Comment by dcapslock** [comment]
`hui-element` applies card_mod. So wrapping in `mod_card` is causing the instability. Below works 100%.

```yaml
type: entities
entities:
  - type: custom:hui-element
    row_type: section
    card_mod:
      debug: true
      style: |
        .divider {
          background-color: red !important;
        }
  - type: custom:hui-element
    row_type: section
    card_mod:
      debug: true
      style: |
        .divider {
          background-color: red !important;
        }
```

**Comment by Mariusthvdb** [comment]
maybe off topic, but we can just do
```
  - type: section
    card_mod:
      style: |
        .divider {
          background: red !important;
        }
```
inside an entities card....

other than that, all of my `custom:hui-element`s with a card_mod style work just fine.
it already was ok in 3.5.0 (retracted) and is in release 4.0.0

**Comment by ildar170975** [comment]
> hui-element applies card_mod. So wrapping in mod_card is causing the instability

Interesting, since when hui-element does not need mod-card? Is it after alpha/beta?

**Comment by dcapslock** [comment]
hui-element has applied card-mod since [Nov 2021](https://github.com/thomasloven/lovelace-hui-element/commit/c712cc65b3e19c878991e5a230df8d7ba86d91f8). To both `element` and `row`. It's noted in the [readme](https://github.com/thomasloven/lovelace-hui-element#a-note-on-card-mod).

> If you are using [card-mod](https://github.com/thomasloven/lovelace-card-mod), hui-element will be "transparent".
>
> I.e. styling hui-element itself should be as if styling the element inside it.

**Comment by ildar170975** [comment]
Disregard my comment - https://github.com/thomasloven/lovelace-card-mod/issues/276#issuecomment-3415942619
I recall that the issue was not about “hui-element requires mod-card”. 
In my examples, hui-element is placed inside mod-card to simulate a case when an element without own ha-card (here - section) is placed inside a decluttering template, and hence requires mod-card. 
So, in this case for a chain “decluttering card - mod-card - hui-section-row” (if I recall the last name properly) the card-mod gives unstable results dependently on whether a line-break is used or not.
If this is resolved in the latest beta - ok. 
If not - please reopen.

**Comment by dcapslock** [comment]
Please submit a new issue with specific config that shows your issue. This issue as listed has been addressed. Thank you.

**Comment by ildar170975** [comment]
The 1st post already has a description of the use case and explanation why hui-element is placed inside mod-card. 
What do you mean by “addressed”? By whom?

If I create a new issue - it will be same, what is the point then?

**Comment by dcapslock** [comment]
Please post yaml config exactly as that which is causing the issue. Your last update mentions decluterring card but I do not see that in the config you provided.

I can only test configs that are provided. Here the config provided has been tested and response provided.
